### PR TITLE
Fix code structure

### PR DESCRIPTION
**Pull request by AliceDmitrieva**
Change code structure and naming methods, change layouts and gitignore file.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,49 @@
...
+
+\.idea/libraries/Gradle__com_android_support_support_core_utils_27_1_0_aar\.xml
+
+\.idea/libraries/Gradle__com_android_support_support_fragment_27_1_0_aar\.xml
+
+\.idea/libraries/Gradle__com_android_support_support_media_compat_27_1_0_aar\.xml
+
+\.idea/libraries/Gradle__com_android_support_support_v4_27_1_0_aar\.xml
+
+\.idea/libraries/Gradle__com_android_support_support_vector_drawable_27_1_0_aar\.xml
```
Comment (oxied) [inline .gitignore]: just ignore .idea/libraries folder at all.

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -5,18 +5,18 @@
...
         android:allowBackup="true"
+
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
-        android:theme="@style/AppTheme">
-        <activity android:name=".activities.MainActivity">
+        android:theme="@style/AppTheme"
+        android:usesCleartextTraffic="true">
```
Comment (oxied) [inline app/src/main/AndroidManifest.xml]: What is the purpose of this attribute and why do you need it?

File: app/src/main/java/com/alicedmitrieva/weatherapp/activities/MainActivity.java [hunk 3]
```diff
@@ -48,6 +50,19 @@ protected void onCreate(Bundle savedInstanceState) {
...
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        getSupportFragmentManager()
+                .putFragment(outState, FRAGMENT_TAG, fragment);
+    }
+
+    @Override
+    protected void onRestoreInstanceState(Bundle savedInstanceState) {
+        super.onRestoreInstanceState(savedInstanceState);
+        fragment = getSupportFragmentManager().findFragmentByTag(FRAGMENT_TAG);
```
Comment (oxied) [inline app/src/main/java/com/alicedmitrieva/weatherapp/activities/MainActivity.java]: Is `onSaveInstanceState` and `onRestoreInstanceState` are necessary?

File: app/src/main/java/com/alicedmitrieva/weatherapp/activities/MainActivity.java [hunk 4]
```diff
@@ -98,16 +113,19 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
-    private void sendRequest() {
-        if (spinner != null) {
-            new RespondWeatherDataTask(MainActivity.this, city).execute();
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        if (requestCode == REQUEST_CODE) {
+            String result = data != null ? data.getStringExtra(getString(R.string.extra_unit)) : null;
+            if ((!currentUnit.equals(result)) && (result != null)) {
```
Comment (oxied) [inline app/src/main/java/com/alicedmitrieva/weatherapp/activities/MainActivity.java]: Remove redundant braces `()`

File: app/src/main/java/com/alicedmitrieva/weatherapp/fragments/MainFragment.java [hunk 5]
```diff
@@ -53,4 +60,10 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
 
         return view;
     }
+
+    @Override
+    public void onViewStateRestored(@Nullable Bundle savedInstanceState) {
+        super.onViewStateRestored(savedInstanceState);
+
+    }
```
Comment (oxied) [inline app/src/main/java/com/alicedmitrieva/weatherapp/fragments/MainFragment.java]: Is this method necessary?

File: .gitignore (hunk 1 above)
Comment (AliceDmitrieva) [reply .gitignore]: Fixed

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (AliceDmitrieva) [reply app/src/main/AndroidManifest.xml]: It uses for Android 9 to have Internet connection. I will add comment to AndroidManifest.

File: app/src/main/java/com/alicedmitrieva/weatherapp/activities/MainActivity.java (hunk 3 above)
Comment (AliceDmitrieva) [reply app/src/main/java/com/alicedmitrieva/weatherapp/activities/MainActivity.java]: We discussed it and decided to leave it.

File: app/src/main/java/com/alicedmitrieva/weatherapp/activities/MainActivity.java (hunk 4 above)
Comment (AliceDmitrieva) [reply app/src/main/java/com/alicedmitrieva/weatherapp/activities/MainActivity.java]: Fixed

File: app/src/main/java/com/alicedmitrieva/weatherapp/fragments/MainFragment.java (hunk 5 above)
Comment (AliceDmitrieva) [reply app/src/main/java/com/alicedmitrieva/weatherapp/fragments/MainFragment.java]: Was removed in PR #3

**Comment by AliceDmitrieva** [comment]
These comments will be fixed in PR #3 cause it is based on the current branch.

**Pull request closed without merging**